src/main/java/org/team1540/robot2024/AlertManager.java

```diff
@@ -0,0 +1,80 @@
+package org.team1540.robot2024;
+
+import com.ctre.phoenix6.CANBus;
+import edu.wpi.first.hal.can.CANStatus;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.Timer;
+import org.team1540.robot2024.util.Alert;
+
+/**
+ * Class for managing high-level robot alerts, such as low battery and CAN bus errors. Subsystem-specific alerts,
+ * such as motor disconnects, should be handled by their respective subsystems.
+ */
+public class AlertManager {
+    private static AlertManager instance;
+
+    public static AlertManager getInstance() {
+        if (instance == null) instance = new AlertManager();
+        return instance;
+    }
+
+    private static final double lowBatteryDisableTime = 1.5;
+    private static final double lowBatteryVoltageThreshold = 12.0;
+    private static final double canErrorTimeThreshold = 0.5;
+
+    private final Timer disabledTimer = new Timer();
+    private final Timer canInitialErrorTimer = new Timer();
+    private final Timer canErrorTimer = new Timer();
+    private final Timer canivoreErrorTimer = new Timer();
+
+    private int lastRioCanTEC = 0;
+    private int lastRioCanREC = 0;
+    private int lastCanivoreTEC = 0;
+    private int lastCanivoreREC = 0;
+
+    private final Alert canErrorAlert =
+            new Alert("RIO CAN bus error, robot functionality may be severely limited", Alert.AlertType.ERROR);
+    private final Alert canivoreErrorAlert =
+            new Alert("Swerve CAN bus error, drive functionality may be severely limited", Alert.AlertType.ERROR);
+    private final Alert lowBatteryAlert =
+            new Alert("Battery voltage is low, consider replacing it", Alert.AlertType.WARNING);
```

Comment:
Long messages take longer to read, and I don't know that they provide a benefit here. "RIO CAN error," and "Swerve CAN error," and "low battery" accomplish the same thing. For battery showing the actual voltage would be nice but otherwise try to keep them short.